sktime/benchmarking/_benchmarking_dataclasses.py

```diff
@@ -15,9 +15,24 @@
 def _coerce_data_for_evaluate(dataset_loader):
     """Coerce data input object to a dict to pass to forecasting evaluate."""
     if callable(dataset_loader) and not hasattr(dataset_loader, "load"):
+        # Case 1: Loader function, e.g., load_longley
         data = dataset_loader()
     elif callable(dataset_loader) and hasattr(dataset_loader, "load"):
-        data = dataset_loader.load()
+        # Case 2: Dataset class, e.g., Longley
+        X = dataset_loader().load("X")
+        y = dataset_loader().load("y")
+        if "dataset_forecasting" in dataset_loader().get_tags()["object_type"]:
```

Comment:
1. I think this introduces too strong coupling and is inconsistent with case 1. Think about what happens if you add a new type of dataset, you have to add a new sub-branch here. We should think about how to remove this coupling.

2. I think the condition is not correct, since the `"object_type"` tag can be `str` or list of `str`. If we do this, then we should use the `is_scitype` utility.